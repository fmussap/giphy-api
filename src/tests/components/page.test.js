import React from 'react'
import { shallow } from 'enzyme'
import Page from '../../components/pagination'

let spy, wrapper
beforeEach(() => {
  spy = jest.fn()
  wrapper = shallow(<Page
    onClick={spy}
    page={1}
    pageLink={'a'}
  />)
})

test('should render Page correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should handle onClick', () => {
  wrapper.find('Page').at(0).simulate('click')
  expect(spy).toHaveBeenLastCalledWith(1)
})
