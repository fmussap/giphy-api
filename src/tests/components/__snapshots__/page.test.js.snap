// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page correctly 1`] = `
<ul
  className="pagination"
>
  <Page
    className="prev"
    onClick={[Function]}
    page="<"
    pageLink="a"
  />
  <div
    className="pagination-item active"
    key="0"
  >
    <Page
      onClick={[Function]}
      page={1}
      pageLink="a"
    />
  </div>
  <Page
    className="next"
    onClick={[Function]}
    page=">"
    pageLink="a"
  />
</ul>
`;
